lib/profileAgents.js

```diff
@@ -0,0 +1,361 @@
+    fields: {'profileAgent.account': 1, 'profileAgent.id': 1},
+    options: {
+      partialFilterExpression: {account: {$exists: true}},
+      unique: true,
+      background: false
+    }
+  }]);
+});
+
+/**
+ * Creates a ProfileAgent and associates it with an account if provided.
+ *
+ * @param {String} accountId the id of the account to associate with the
+ *                           ProfileAgent.
+ *
+ * @return {Promise<ProfileAgent>} resolves to a ProfileAgent.
+ */
+exports.create = async ({accountId} = {}) => {
+  // 1. Generate a random secret
+  const secret = base64url.encode(crypto.randomBytes(32));
+  const handle = 'primary';
+  // 2. Generate capability agent for the zCap key
+  const capabilityAgent = await CapabilityAgent.fromSecret({handle, secret});
+  // 3. Create keystore in order to create the zCap key for the profile agent
+  // TODO: Will want to store the capability agent up here first, otherwise we
+  //       have no clean way of cleaning up the other stuff that gets created
+  //       on other systems should we experience a failure.
+  const keystore = await kms.createKeystore(
+    {capabilityAgent, referenceId: 'primary'});
+  // 4. Create the zCap key for the profile agent
+  const keystoreAgent = kms.getKeystoreAgent({capabilityAgent, keystore});
+  const key = await keystoreAgent.generateKey(
+    {type: 'Ed25519VerificationKey2018', kmsModule: KMS_MODULE});
+  // 5. Use the zCap key to create the DID for the profile agent
+  await utils.setKeyId({key});
+  // 6. Generate capability agent for ProfileAgent
+  const paZcapAgent = new CapabilityAgent({handle: 'primary', signer: key});
+  // 7. Create keystore to store keys controlled by the profile agent
+  const paKeystore = await kms.createKeystore(
+    {capabilityAgent: paZcapAgent, referenceId: 'primary'});
+  const paKeystoreAgent = kms.getKeystoreAgent(
+    {capabilityAgent: paZcapAgent, keystore: paKeystore});
+
+  const did = key.id.split('#')[0];
+  const profileAgent = {
+    id: did,
+    sequence: 0,
+    account: accountId,
+    controller: {
+      id: capabilityAgent.id,
+      seed: secret,
+      keystore: keystore.id
+    },
+    keystore: paKeystore.id,
+    capabilityInvocationKey: (await key.getKeyDescription()).id
+  };
+  const now = Date.now();
+  const meta = {created: now, updated: now};
+  let record = {
+    meta,
+    profileAgent
+  };
+  try {
+    const collection = utils.getCollection('profile-profileAgent');
+    const result = await collection.insert(record, database.writeOptions);
+    record = result.ops[0];
+  } catch(e) {
+    if(!database.isDuplicateError(e)) {
+      throw e;
+    }
+    throw new BedrockError(
+      'Duplicate profile agent.',
+      'DuplicateError', {
+        public: true,
+        httpStatusCode: 409
+      }, e);
+  }
+  return {
+    ...record,
+    profileAgent: {
+      ...record.profileAgent,
+      capabilityAgent: paZcapAgent,
+      keystoreAgent: paKeystoreAgent
```

Comment:
I worry that mixing the agent stuff in here is going to get messy and goes against our other patterns. Perhaps we should just expose "getAgents" and you pass the data about a profile agent into that.